src/vellum/workflows/state/context.py

```diff
@@ -156,24 +156,64 @@ def resolve_workflow_deployment(
         Returns:
             BaseWorkflow instance if found, None otherwise
         """
-        if not self.generated_files or not self.namespace:
-            return None
+        if self.generated_files and self.namespace:
+            expected_prefix = generate_workflow_deployment_prefix(deployment_name, release_tag)
+            workflow_file_key = f"{expected_prefix}/workflow.py"
+            if workflow_file_key in self.generated_files:
+                try:
+                    from vellum.workflows.workflows.base import BaseWorkflow
+
+                    WorkflowClass = BaseWorkflow.load_from_module(f"{self.namespace}.{expected_prefix}")
+                    workflow_instance = WorkflowClass(context=WorkflowContext.create_from(self), parent_state=state)
+                    return workflow_instance
+                except Exception:
+                    pass
+
+        # Try using the pull API to fetch the workflow deployment
+        try:
+            import io
+            import zipfile
 
-        expected_prefix = generate_workflow_deployment_prefix(deployment_name, release_tag)
+            from vellum_ee.workflows.server.virtual_file_loader import VirtualFileFinder
 
-        workflow_file_key = f"{expected_prefix}/workflow.py"
-        if workflow_file_key not in self.generated_files:
-            return None
+            response = self.vellum_client.workflows.pull(
+                deployment_name, request_options={"additional_query_parameters": {"release_tag": release_tag}}
+            )
 
-        try:
-            from vellum.workflows.workflows.base import BaseWorkflow
+            zip_bytes = b"".join(response)
+            zip_buffer = io.BytesIO(zip_bytes)
+
+            pulled_files = {}
+            with zipfile.ZipFile(zip_buffer) as zip_file:
+                for file_name in zip_file.namelist():
+                    with zip_file.open(file_name) as source:
+                        content = source.read().decode("utf-8")
+                        pulled_files[file_name] = content
+
+            # Create a unique namespace for this deployment
+            expected_prefix = generate_workflow_deployment_prefix(deployment_name, release_tag)
+            deployment_namespace = f"deployment_{expected_prefix}"
+
+            finder = VirtualFileFinder(pulled_files, deployment_namespace)
+            import sys
+
+            sys.meta_path.append(finder)
```

Comment:
instead of adding VirtualFileFinder, we should instead update self.generated_files with the new files we pulled